server/server.go

```diff
@@ -110,7 +111,13 @@ func errorEncoder(ctx context.Context, err error, w http.ResponseWriter) {
 	rErr := err.(microserver.ResponseError)
 	uErr := rErr.Underlying()
 
-	rErr.SetCode(microserver.CodeInternalError)
-	rErr.SetMessage(uErr.Error())
-	w.WriteHeader(http.StatusInternalServerError)
+	if creator.IsAlreadyExists(uErr) {
+		rErr.SetCode(microserver.CodeResourceAlreadyExists)
+		rErr.SetMessage(uErr.Error())
+		w.WriteHeader(http.StatusConflict)
+	} else {
+		rErr.SetCode(microserver.CodeInternalError)
+		rErr.SetMessage(uErr.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+	}
```

Comment:
I am not sure about all the details now but you might need to write the response body here. Uncertain. See https://github.com/giantswarm/api/blob/40bd6e13cfb6a0176d9ab72477546522437890cd/server/server.go#L321-L325.